src/Shared/ServerInfrastructure/BufferExtensions.cs

```diff
@@ -129,47 +129,42 @@ internal static void WriteAscii(ref this BufferWriter<PipeWriter> buffer, string
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    internal static unsafe void WriteNumeric(ref this BufferWriter<PipeWriter> buffer, ulong number)
+    internal static void WriteNumeric(ref this BufferWriter<PipeWriter> buffer, ulong number)
     {
         const byte AsciiDigitStart = (byte)'0';
 
-        var span = buffer.Span;
-        var bytesLeftInBlock = span.Length;
+        var start = buffer.Span;
 
         // Fast path, try copying to the available memory directly
         var simpleWrite = true;
-        fixed (byte* output = span)
+        if (number < 10 && start.Length >= 1)
         {
-            var start = output;
-            if (number < 10 && bytesLeftInBlock >= 1)
-            {
-                *(start) = (byte)(((uint)number) + AsciiDigitStart);
-                buffer.Advance(1);
-            }
-            else if (number < 100 && bytesLeftInBlock >= 2)
-            {
-                var val = (uint)number;
-                var tens = (byte)((val * 205u) >> 11); // div10, valid to 1028
+            start[0] = (byte)(((uint)number) + AsciiDigitStart);
+            buffer.Advance(1);
+        }
+        else if (number < 100 && start.Length >= 2)
+        {
+            var val = (uint)number;
+            var tens = (uint)(byte)((val * 205u) >> 11); // div10, valid to 1028
 
-                *(start) = (byte)(tens + AsciiDigitStart);
-                *(start + 1) = (byte)(val - (tens * 10) + AsciiDigitStart);
-                buffer.Advance(2);
-            }
-            else if (number < 1000 && bytesLeftInBlock >= 3)
-            {
-                var val = (uint)number;
-                var digit0 = (byte)((val * 41u) >> 12); // div100, valid to 1098
-                var digits01 = (byte)((val * 205u) >> 11); // div10, valid to 1028
-
-                *(start) = (byte)(digit0 + AsciiDigitStart);
-                *(start + 1) = (byte)(digits01 - (digit0 * 10) + AsciiDigitStart);
-                *(start + 2) = (byte)(val - (digits01 * 10) + AsciiDigitStart);
-                buffer.Advance(3);
-            }
-            else
-            {
-                simpleWrite = false;
-            }
+            start[0] = (byte)(tens + AsciiDigitStart);
+            start[1] = (byte)(val - (tens * 10) + AsciiDigitStart);
+            buffer.Advance(2);
+        }
+        else if (number < 1000 && start.Length >= 3)
+        {
+            var val = (uint)number;
+            var digit0 = (uint)(byte)((val * 41u) >> 12); // div100, valid to 1098
+            var digits01 = (uint)(byte)((val * 205u) >> 11); // div10, valid to 1028
+
+            start[0] = (byte)(digit0 + AsciiDigitStart);
+            start[1] = (byte)(digits01 - (digit0 * 10) + AsciiDigitStart);
+            start[2] = (byte)(val - (digits01 * 10) + AsciiDigitStart);
+            buffer.Advance(3);
+        }
+        else
+        {
+            simpleWrite = false;
```

Comment:
You could get rid of `simpleWrite` variable altogether? It's used only for
```suggestion
            WriteNumericMultiWrite(ref buffer, number);
```
which can be done directly.

Reply:
removed